drape/harfbuzz_shaping.cpp

```diff
@@ -82,8 +83,9 @@ size_t ScriptSetIntersect(char32_t codepoint, TScriptsArray & inOutScripts, size
   // Intersect both script sets.
   ASSERT(!contains(USCRIPT_INHERITED), ());
   size_t outSize = 0;
-  for (auto const currentScript : inOutScripts)
+  for (size_t i = 0; i < inOutScriptsCount; ++i)
```

Comment:
?

Reply:
This is the fix. The previous code always iterated the whole (even not initialized) array instead of only a few initialized entries.